src/dodal/devices/areadetector/andor2_epics/andor2_io.py

```diff
@@ -0,0 +1,53 @@
+from enum import Enum
+
+from ophyd_async.epics.adcore._core_io import ADBaseIO
+from ophyd_async.epics.signal import (
+    epics_signal_r,
+    epics_signal_rw,
+    epics_signal_rw_rbv,
+)
+
+
+class Andor2TriggerMode(str, Enum):
+    internal = "Internal"
+    ext_trigger = "External"
+    ext_start = "External Start"
+    ext_exposure = "External Exposure"
+    ext_FVP = "External FVP"
+    soft = "Software"
+
+
+class ImageMode(str, Enum):
+    single = "Single"
+    multiple = "Multiple"
+    continuous = "Continuous"
+    fast_kinetics = "Fast Kinetics"
+
+
+class ADBaseDataType(str, Enum):
+    UInt16 = "UInt16"
+    UInt32 = "UInt32"
+    b1 = ""
+    b2 = ""
+    b3 = ""
+    b4 = ""
+    b5 = ""
+    b6 = ""
```

Comment:
Should: I don't think you need all these empty string ones, it should be fine without them

Reply:
It fail the data type check when I try to connect without matching them. 
![image](https://github.com/user-attachments/assets/fa2c030f-9ab4-4bcd-a94d-3489d8768432)

Reply:
Odd, from https://github.com/bluesky/ophyd-async/blob/55f3552e667569d872d4f4b5ffc1df4d64601a17/src/ophyd_async/epics/signal/_aioca.py#L230 it looks like the options should be filtered to unique ones. For reference the PV:
```
[ffv81422@p99-ws001 ~]$ caget BL99P-EA-DET-03:CAM:DataType_RBV
BL99P-EA-DET-03:CAM:DataType_RBV UInt16
[ffv81422@p99-ws001 ~]$ caget -d31 BL99P-EA-DET-03:CAM:DataType_RBV
BL99P-EA-DET-03:CAM:DataType_RBV
    Native data type: DBF_ENUM
    Request type:     DBR_CTRL_ENUM
    Element count:    1
    Value:            UInt16
    Status:           NO_ALARM
    Severity:         NO_ALARM
    Enums:            (10)
                      [ 0] UInt16
                      [ 1] UInt32
                      [ 2] 
                      [ 3] 
                      [ 4] 
                      [ 5] 
                      [ 6] 
                      [ 7] 
                      [ 8] Float32
                      [ 9] Float64
```
@coretl, is this a bug in ophyd-async?

Reply:
Does it go away if you add `"_": ""` to the enum members? The error is coming from  [here](https://github.com/bluesky/ophyd-async/blob/55f3552e667569d872d4f4b5ffc1df4d64601a17/src/ophyd_async/epics/signal/_common.py#L45) which seems to do a set

Reply:
Yes it work with `"_" : "" `, as long as we have an empty string, I guess the set on the right [here](https://github.com/bluesky/ophyd-async/blob/55f3552e667569d872d4f4b5ffc1df4d64601a17/src/ophyd_async/epics/signal/_common.py#L45) will have one "" and we need to match it to pass the check. Would it be better to .remove it in ophyd or add a blank when we define the data class?

Reply:
I'm not sure either is particularly good, but that's what EPICS gives us. And it's technically possible for the PV to be set to "", so I think adding it to the Enum subclass is more technically correct.